app/views/dashboard/index.html.erb

```diff
@@ -11,15 +11,20 @@
     <div id='dashboard' class="gridster col-md-12">
       <ul>
         <% User.current.widgets.each do |widget| %>
-          <%= content_tag(:li, widget_data(widget)) do %>
-            <div class='widget_control'>
-              <a class='remove' data-url='<%= widget_path(widget) %>'>&times;</a>
-              <a class='minimize'>&minus;</a>
-            </div>
-            <div class="widget <%= widget.name.parameterize %>">
-              <%= render_widget(widget) %>
-            </div>
-          <% end %>
+            <%= content_tag(:li, widget_data(widget)) do %>
+              <div class='widget_control'>
+                <a class='remove' data-url='<%= widget_path(widget) %>'>&times;</a>
+                <a class='minimize'>&minus;</a>
+              </div>
+              <div class="widget <%= widget.name.parameterize %>">
+
+                <%= begin
+                      render_widget(widget)
+                    rescue ActionView::MissingTemplate => e; e.class
```

Comment:
I think that the begin/rescue clause should be in the render_widget helper method. this has no place in a view.
also, something seems funky with the indentation...